modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/HelpText/HelpText.es.js

```diff
@@ -0,0 +1,259 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayButton from '@clayui/button';
+import {usePrevious} from '@liferay/frontend-js-react-web';
+import classNames from 'classnames';
+import React, {useEffect, useState} from 'react';
+
+import {transformData} from './transform.es';
+
+const HelpText = ({
+	defaultLanguageId,
+	displayErrors,
+	editingLanguageId,
+	errorMessage: initialErrorMessage,
+	label,
+	name,
+	parameter: initialParameter,
+	readOnly,
+	repeatable,
+	required,
+	selectedValidation: initialSelectedValidation,
+	showLabel = true,
+	style,
+	tooltip,
+	valid,
+	value,
+	visible,
+}) => {
+	const [{errorMessage}, setState] = useState({
+		errorMessage: initialErrorMessage,
+		parameter: initialParameter,
+		selectedValidation: initialSelectedValidation,
+	});
+
+	const renderLabel =
+		(label && showLabel) || required || tooltip || repeatable;
+
+	const fieldDetailsId = name + '_fieldDetails';
+
+	const [enableHelpText, setEnableHelpText] = useState(false);
+	const [showHideButton, setShowHideButton] = useState('Show more');
+
+	const hasError = displayErrors && errorMessage && !valid;
+
+	const prevEditingLanguageId = usePrevious(editingLanguageId);
+
+	useEffect(() => {
+		if (prevEditingLanguageId !== editingLanguageId) {
+			setState((prevState) => {
+				const {errorMessage = {}, parameter = {}} = value;
+
+				return {
+					...prevState,
+					errorMessage:
+						errorMessage[editingLanguageId] !== undefined
+							? errorMessage[editingLanguageId]
+							: errorMessage[defaultLanguageId],
+					parameter:
+						parameter[editingLanguageId] !== undefined
+							? parameter[editingLanguageId]
+							: parameter[defaultLanguageId],
+				};
+			});
+		}
+	}, [defaultLanguageId, editingLanguageId, prevEditingLanguageId, value]);
```

Comment:
@mateusgomes01 Please remove all the code you don’t actually need in the `Help Text` field. For example, this `useEffect` and the `const [{errorMessage}, setState] = useState` it's a copy from the `Validation` field, I don't think we need this here, what do you think? Also, I believe that the `hasError` variable it's not necessary here (it's also a copy from the `Validation` field).

Reply:
Ok!